specification/configuration.md

```diff
@@ -130,6 +136,27 @@ beyond the OpenTelemetry specification exist.
   - Distribution MUST default to `otlp`
   - Distribution MUST offer `jaeger-thrift-splunk` that defaults to `http://127.0.0.1:9080/v1/trace`
 
+### Real User Monitoring Libraries
+
+Real User Monitoring (RUM) instrumentation libraries cannot use environment variables for configuration.
+Instead, they MUST expose a `SplunkRum` object/class/namespace (depending on the language used) that allows
+setting the following properties (in a language-specific way: Java may use builders, Swift can use named
+parameters with default values, JavaScript can use objects, etc.):
+
+| Configuration property (default value) | Description                                                                                                                                                                                                 |
+| -------------------------------------- | -----------                                                                                                                                                                                                 |
+| `beaconUrl` ()                         | RUM beacon URL, ex. `https://rum-ingest.<realm>.signalfx.com/v1/rum`. [1]                                                                                                                                   |
+| `rumAccessToken` ()                    | RUM authentication token. [1]                                                                                                                                                                               |
+| `applicationName` ()                   | Instrumented application name. [1]                                                                                                                                                                          |
```

Comment:
We use the term `app` for this.

Reply:
on the programmatic configuration? not `appName` ?

Reply:
Yep - https://github.com/signalfx/splunk-otel-js-web/blob/main/docs/Configuration.md#general-configuration-options - not built in iOS yet but let's be consistent.  In general I like it when the config name matches up with the attribute name.

Reply:
hmm. can't say that I'm a fan of cryptic configuration method names.

Reply:
Connected to https://github.com/signalfx/gdi-specification/pull/112#discussion_r657296227: should we continue to use `app`/`appName`/`applicationName`? `serviceName` & `service.name` would be more consistent with otel conventions

Reply:
Like @flands said, the user should only have to set one thing (IMO the application name, whatever we call that configuration item), and the library should do the needful under the hood and set both `app` on spans, and `service.name` on the Resource to that value.